src/pybind/mgr/dashboard/controllers/osd.py

```diff
@@ -56,7 +56,21 @@ def get_osd_map(self):
 
     @handle_send_command_error('osd')
     def get(self, svc_id):
-        histogram = CephService.send_command('osd', srv_spec=svc_id, prefix='perf histogram dump')
+        """
+        Returns collected data about an OSD.
+
+        :return: Returns the requested data. The `histogram` key man contain a
+                 string with an error that occurred when the OSD is down.
+        """
+        try:
+            histogram = CephService.send_command('osd', srv_spec=svc_id,
+                                                 prefix='perf histogram dump')
+        except SendCommandError as e:
+            if 'osd down' in e.message:
+                histogram = e.message
```

Comment:
i guess this breaks the test of http://pulpito.ceph.com/kchai-2018-09-11_09:51:05-rados-wip-kefu-testing-2018-09-10-1219-distro-basic-mira/3005446/ .

```
2018-09-11T10:14:52.021 INFO:tasks.cephfs_test_runner:Starting test: test_safe_to_destroy (tasks.mgr.dashboard.test_osd.OsdTest)
2018-09-11T10:14:52.021 INFO:tasks.mgr.dashboard.helper:request GET to https://mira071.front.sepia.ceph.com:7789/api/osd/5/safe_t
o_destroy
2018-09-11T10:14:52.028 INFO:tasks.ceph.mgr.x.mira071.stdout:::ffff:172.21.0.51 - - [11/Sep/2018:10:14:52] "GET /api/osd/5/safe_t
o_destroy HTTP/1.1" 200 25 "" "python-requests/2.12.5"
2018-09-11T10:14:52.029 INFO:tasks.mgr.dashboard.helper:request GET to https://mira071.front.sepia.ceph.com:7789/api/osd/0/safe_t
o_destroy
2018-09-11T10:14:52.036 INFO:tasks.ceph.mgr.x.mira071.stderr:2018-09-11 10:14:52.034 7f6f38515700 -1 mgr.server reply reply (16)
Device or resource busy OSD(s) 0 last reported they still store some PG data, and not all PGs are active+clean; we cannot be sure
 they aren't still needed.
2018-09-11T10:14:52.069 INFO:tasks.cephfs_test_runner:test_safe_to_destroy (tasks.mgr.dashboard.test_osd.OsdTest) ... ERROR
```
```
2018-09-11T10:14:52.080 INFO:tasks.cephfs_test_runner:ERROR: test_safe_to_destroy (tasks.mgr.dashboard.test_osd.OsdTest)
2018-09-11T10:14:52.080 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2018-09-11T10:14:52.080 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2018-09-11T10:14:52.081 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-kefu-testing-2018-09-
10-1219/qa/tasks/mgr/dashboard/test_osd.py", line 109, in test_safe_to_destroy
2018-09-11T10:14:52.081 INFO:tasks.cephfs_test_runner:    self.wait_until_equal(get_destroy_status, False, 10)
2018-09-11T10:14:52.081 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-kefu-testing-2018-09-
10-1219/qa/tasks/ceph_test_case.py", line 118, in wait_until_equal
2018-09-11T10:14:52.081 INFO:tasks.cephfs_test_runner:    val = get_fn()
2018-09-11T10:14:52.081 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-kefu-testing-2018-09-10-1219/qa/tasks/mgr/dashboard/test_osd.py", line 108, in get_destroy_status
2018-09-11T10:14:52.081 INFO:tasks.cephfs_test_runner:    return self.jsonBody()['safe-to-destroy']
2018-09-11T10:14:52.081 INFO:tasks.cephfs_test_runner:KeyError: 'safe-to-destroy'
```

Reply:
@tchaikov @p-na is already working on a fix for that in #24024

Reply:
No, [PR#23491](https://github.com/ceph/ceph/pull/23491/files#diff-8737c8033daa080af84b77f5308abcddR108) makes the QA flaky and as it's already merged, I've created [PR#24024](https://github.com/ceph/ceph/pull/24024) to fix it. This PR should be fine.